### PR TITLE
Dependency fixes

### PR DESCRIPTION
**Pull request by evilstreak**
Some miscellaneous issues to get my dev environment (mostly) working again.

### CODE REVIEW

**Comment by evilstreak** [comment]
Dang, the last version of numpy to support Python 3.7 is 1.21.6. Going to need to rethink that change.

**Comment by nickstenning** [comment]
> Dang, the last version of numpy to support Python 3.7 is 1.21.6. Going to need to rethink that change.

We can probably get away with this as `requirements-dev.txt` is only used in our dev and CI environments:

```
numpy==1.21.6; python_version == '3.7'
numpy==1.24.1; python_version > '3.7'
```

(Requires `pip>=6`)

**Comment by evilstreak** [comment]
> ```
> numpy==1.21.6; python_version == '3.7'
> numpy==1.24.1; python_version > '3.7'
> ```

TIL, thanks!

File: requirements-dev.txt [hunk 1]
```diff
@@ -2,7 +2,8 @@ build==0.9.0
 hypothesis==6.56.4
 httpx==0.23.1
 mypy==0.990
-numpy==1.21.5
+numpy==1.21.6; python_version == '3.7'
```
Comment (zeke) [inline requirements-dev.txt:5]: Today I also learned.

https://peps.python.org/pep-0508/#environment-markers

**Review by zeke — APPROVED** [review]

**Review by nickstenning — APPROVED** [review]

File: python/tests/server/test_http.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 from PIL import Image
 import pytest
+import unittest.mock as mock
```
Comment (nickstenning) [inline python/tests/server/test_http.py:6]: This is a slightly unusual way of saying `from unittest import mock` but not a big deal in test code :)

**Pull request merged**